clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignorTest.java

```diff
@@ -105,12 +106,16 @@ public void testOnlyAssignsPartitionsFromSubscribedTopics() {
         String otherTopic = "other";
 
         Map<String, Integer> partitionsPerTopic = new HashMap<>();
-        partitionsPerTopic.put(topic, 3);
-        partitionsPerTopic.put(otherTopic, 3);
-        subscriptions = Collections.singletonMap(consumerId, new Subscription(topics(topic)));
```

Comment:
This test was also starting with an illegal state -- `partitionsPerTopic` only contains metadata for topics included in the subscription. I noticed that we don't seem to be testing the actual valid case, where some consumers have `ownedPartitions` which are no longer in the subscription, so I just adapted this test for the related purpose